app/views/csvfiles/new.html.erb

```diff
@@ -0,0 +1,114 @@
+<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-3-typeahead/4.0.2/bootstrap3-typeahead.js"></script>
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/PapaParse/4.6.3/papaparse.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.7.3/Chart.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/plotly.js/1.33.1/plotly-basic.min.js"></script>
+<link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/css/all.min.css">
+<link href="https://gitcdn.github.io/bootstrap-toggle/2.2.2/css/bootstrap-toggle.min.css" rel="stylesheet">
+<script src="https://gitcdn.github.io/bootstrap-toggle/2.2.2/js/bootstrap-toggle.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/FileSaver.js/1.3.3/FileSaver.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/xlsx/0.14.3/xlsx.full.min.js"></script>
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/css/all.min.css">
+<%= javascript_include_tag('/lib/simple-data-grapher/dist/PublicLab.Grapher.js')%>
+<%= stylesheet_link_tag '/lib/simple-data-grapher/examples/upload_file.css'%>
+<div id="first"></div>
+<script>
+    a = new SimpleDataGrapher("first");
+    var value='<%= current_user %>';
+    <% if current_user %>
+        a.view.createButtons("yes");
+        console.log("logged in");
+        console.log(a.view);
+        var r=a.view.elementId+"_save_CSV";
+        var prev=a.view.elementId+"_prev_file";
+        var saveFlag=false;
+        $("#"+r).click(function(){
+            console.log("save it");
+            saveFlag=true;
+        });
+        $("#"+a.view.plotGraphId).click(function(){
+            console.log("plotted");
+            console.log(a.view.fileTitle,a.view.fileDescription,"inn");
+            if (saveFlag){
+                var arr={};
+                arr["completeCsvMatrix"]=a.view.csvParser.completeCsvMatrix;
+                arr["csvHeaders"]=a.view.csvParser.csvHeaders;
+                arr["csvSampleData"]=a.view.csvParser.csvSampleData;
+                arr["csvValidForYAxis"]=a.view.csvParser.csvValidForYAxis;
+                arr["completeCsvMatrixTranspose"]=a.view.csvParser.completeCsvMatrixTranspose;
+                let csvStringMatrix=a.view.csvParser.completeCsvMatrixTranspose;
+                let csvStringForDownload=encodeURI("data:text/csv;charset=utf-8," + csvStringMatrix.map(e => e.join(",")).join("\n"));
+                $.ajax({
```

Comment:
No error handling in ajax ? Addition of this is required.
this will help prepare us for worst case scenarios. API calls aren't always a success